### ISSUE TITLE
completion of non valid clojure code

### ISSUE THREAD

**Issue by ericdallo** [clojure-lsp/clojure-lsp#270]
For a sample code of:
```clojure
(ns a)
(def bar 1)
```
```clojure
(ns b (:require [a :as foo]))

foo/
```
ATM we don't support completion of `foo/` only `foo/b` because of two issues:

- [x] it's not possible to parse a invalid code with rewrite-clj `z/of-string`. https://github.com/clj-commons/rewrite-clj/issues/87 (workaround fix [here](https://github.com/clojure-lsp/clojure-lsp/commit/52510164a3765d373763485a4862d4c493641a2d#diff-897a27d95724c2a85947bfa6bc7b5ac1086272aeb125972c1b57749beea71050R60-R73))
- [x] we use clj-kondo analysis of current document to find the alias ns, but when the code is invalid, clj-kondo doesn't return any analysis for that file. https://github.com/clj-kondo/clj-kondo/issues/1146

**Comment by bpringe** [comment]
Just want to point out that in the case of a non-valid form, other lsp features temporarily "break." In the gif below you can see when I type `str/` the code lenses disappear and the linting problems disappear except for the invalid form one. Maybe the linting part isn't a big deal, since it's telling you where the problem is. The other symptoms may not be a huge deal either since they go away when the form is valid again.

![completion-non-valid](https://user-images.githubusercontent.com/12722744/107293327-45f34280-6a20-11eb-87c1-7823a572a862.gif)

**Comment by ericdallo** [comment]
Yeah, that's part of clj-kondo issue, since it can't scan a invalid code, so clojure-lsp end updating the current features with no analysis data, we could do not update it when it's a invalid code but I'd prefer making clj-kondo work with that

**Comment by borkdude** [comment]
The clj-kondo issue is now fixed on master.

**Comment by ericdallo** [comment]
Fixed 🎉 Available on next release

**Comment by ericdallo** [comment]
@bpringe could you test it with Calva if it works for you too?

**Comment by bpringe** [comment]
It works on the first typing of `/` and then every other time after that. I don't see the request being sent on the times it doesn't work, so maybe it's just something that needs to be configured in the client lib we use (or something is broken).

![autocomplete-lsp](https://user-images.githubusercontent.com/12722744/108535781-89577780-7290-11eb-8d04-47d7bb2d92e1.gif)

This case that it doesn't work is only if right after it works I backspace and type `/` again, but if I move the cursor elsewhere and back between tries, it works every time.

I wonder if the `triggerCharacters` property of the `CompletionItems` interface in the spec (server capability) could help here, but maybe this is purely a client issue.

There's also this case I found in which it doesn't work ever. If it just showed completions for `edn` when I typed `n` then it doesn't show the completions when I type the `/`.

![autocomplete-lsp2](https://user-images.githubusercontent.com/12722744/108537874-f966fd00-7292-11eb-9eed-770b9b5b95ac.gif)

--------

Also, if you noticed double completions in the gif, `edn` is an alias to `clojure.edn` - this is in a .clj file, and one completion item is coming from the clojure jar, and the other from the clojurescript jar. Should clojurescript completion items be showing in a .clj file?

For example, for the `clojure.edn/read` completion items, one is from:
```
/home/brandon/.m2/repository/org/clojure/clojure/1.10.0/clojure-1.10.0.jar:clojure/edn.clj
```
and the other is from:
```
/home/brandon/.m2/repository/org/clojure/clojurescript/1.10.520/clojurescript-1.10.520.jar:clojure/edn.cljs
```
